src/main/java/io/strimzi/kafka/bridge/SourceBridgeEndpoint.java

```diff
@@ -82,22 +79,33 @@ protected void handleClose() {
     /**
      * Send a record to Kafka
      *
-     * @param krecord   Kafka record to send
-     * @param handler   handler to call if producer with unsettled is used
+     * @param record Kafka record to send
+     * @param isAsync if the send doesn't need to get metadata via the Kafka related callback
+     * @return a CompletionStage bringing the metadata
      */
-    protected void send(KafkaProducerRecord<K, V> krecord, Handler<AsyncResult<RecordMetadata>> handler) {
-
-        log.debug("Sending record {}", krecord);
-        if (handler == null) {
-            this.producer.send(krecord);
+    protected CompletionStage<RecordMetadata> send(ProducerRecord<K, V> record, boolean isAsync) {
```

Comment:
Having two different methods could be a good idea.

> It would also allow you to document on which thread the returned CompletionStage gets completed on.

Do you mean highlighting it's happening in the producer IO thread?

Reply:
Yes

Reply:
While I liked the idea of having two separate send methods for clarity, it remembered me this issue I discovered and raised here https://github.com/strimzi/strimzi-kafka-bridge/issues/730

In the end. With `async=true` we can reply to the HTTP client straight away not waiting for the Kafka send to complete but in reality we still need the callback in place for closing the HTTP send span at the right time (as explained in the issue).
With the `async=false` we don't send the HTTP response until the send callback is done (in order to get the metadata) and the span is closed.

I have a got a fix locally for the Vert.x based version and haven't fixed it on this PR yet because not sure I should have done them together which could make sense at this point. It means that the `isAsync` could be not needed in the `SourceBridgeEndpoint.send` method but it could be used at upper level and maybe being called `isIgnoringResult`. Something I have to think of anyway.